test/e2e/e2e.go

```diff
@@ -360,9 +360,103 @@ var _ = ginkgo.Describe("[efs-csi] EFS CSI", func() {
 			encryptInTransit := false
 			testEncryptInTransit(f, &encryptInTransit)
 		})
+
+		ginkgo.It("should create an access point with the desired subpath pattern", func() {
+
+			ginkgo.By("Creating EFS Storage Class, PVC and associated PV")
+			pvName := "foo"
+			pvcName := "bar"
+			directoryCreated := fmt.Sprintf("/%s/%s", pvName, pvcName)
+			params := map[string]string{
+				"provisioningMode": "efs-ap",
+				"fileSystemId":     FileSystemId,
+				"directoryPerms":   "777",
+				"subPathPattern":   "${.PV.name}/${.PVC.name}",
+				"pvName":           pvName,
+				"pvcName":          pvcName,
+			}
+
+			sc := GetStorageClass(params)
+			sc, err := f.ClientSet.StorageV1().StorageClasses().Create(context.TODO(), sc, metav1.CreateOptions{})
+			framework.ExpectNoError(err, "creating storage class")
+			pvc, err := createEFSPVCPVDynamicProvisioning(f.ClientSet, f.Namespace.Name, pvcName, sc.Name)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Deploying a pod that applies the PVC to create an access point")
+			pod := e2epod.MakePod(f.Namespace.Name, nil, []*v1.PersistentVolumeClaim{pvc}, false, "while :; do echo '.'; sleep 5 ; done")
+			pod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(context.TODO(), pod, metav1.CreateOptions{})
+			framework.ExpectNoError(err, "creating pod")
+			framework.ExpectNoError(e2epod.WaitForPodNameRunningInNamespace(f.ClientSet, pod.Name, f.Namespace.Name), "waiting for pod running")
+			defer func() {
+				_ = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
+				_ = f.ClientSet.CoreV1().PersistentVolumeClaims(f.Namespace.Name).Delete(context.TODO(), pvc.Name, metav1.DeleteOptions{})
+			}()
+
+			// podClient := f.PodClientNS(f.Namespace.Name)
+			// nodePods, err := podClient.List(context.Background(), metav1.ListOptions{
+			// 	LabelSelector: "app=efs-csi-node",
+			// })
+			// framework.ExpectNoError(err, "Fetching efs-csi-driver node pods")
+			// framework.ExpectNotEqual(len(nodePods.Items), 0, "Get node pods count")
+			// nodePodName := nodePods.Items[0].Name
+			ginkgo.By("Checking if access point's directory matches expected structure") //where can i check that the directory was created??
+			cmd := fmt.Sprintf("cd mnt && cd %s && pwd", directoryCreated)
```

Comment:
accesspoint mountPath will be `/mnt/volume1`
So check for the subdirectories inside this folder.
You can refer [here](https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/master/vendor/k8s.io/kubernetes/test/e2e/framework/pod/create.go#L224)